pkg/testutils/e2e/cluster/cluster.go

```diff
@@ -208,9 +232,69 @@ func BuildE2eCluster(e2eConfig E2eClusterConfig, ops ...E2eClusterOption) (E2eCl
 	return cluster, nil
 }
 
+// merge options from environment variables
+func mergeEnvVariables(config E2eClusterConfig) E2eClusterConfig {
+	config.AutoCleanup = utils.GetBooleanEnvVar("E2E_AUTOCLEANUP", config.AutoCleanup)
+	config.Reuse = utils.GetBooleanEnvVar("E2E_REUSE", config.Reuse)
+	return config
+}
+
+// BuildE2eCluster builds a cluster for e2e tests
+func BuildE2eCluster(
+	t *testing.T,
+	e2eConfig E2eClusterConfig,
+	ops ...E2eClusterOption,
+) (e2ec E2eCluster, err error) {
+	// apply option functions
+	for _, option := range ops {
+		e2eConfig, err = option(e2eConfig)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	e2eConfig = mergeEnvVariables(e2eConfig)
+
+	defer func() {
+		if e2ec != nil && e2eConfig.AutoCleanup {
+			t.Cleanup(func() {
+				_ = e2ec.Delete()
+			})
+		}
+	}()
```

Comment:
This seems like a good place for letting the user know what happened with the e2e cluster if there was an error. We could do something like
```suggestion
	defer func() {
		if e2ec == nil {
			return
		}

		t.Cleanup(func() {
			if t.Failed() && e2eConfig.AutoCleanup {
				t.Logf("Cleaning up failed test cluster. To keep the cluster for debugging, run with E2E_AUTOCLEANUP=1")
			}

			if t.Failed() && !e2eConfig.AutoCleanup {
				t.Logf("Keeping failed test cluster. After debugging, it can be deleted with `kind delete cluster -n %s'", e2ec.Name())
			}

			if e2eConfig.AutoCleanup {
				_ = e2ec.Delete()
				return
			}
		})
	}()
```

Reply:
I think we should review the logging (or lack thereof) in a follow-up issue/PR to ensure consistency.
Regarding giving specific instructions about what to do I think this should be in the documentation, not the code.